config/load.go

```diff
@@ -18,6 +19,9 @@ func Load(configFile string) (*Config, error) {
 	if err != nil {
 		return nil, err
 	}
+	if (config.HttpConfig.Port < 0) || (config.HttpConfig.Port > 65535) || (config.HttpsConfig.Port < 0) || (config.HttpsConfig.Port > 65535) {
```

Comment:
This is a utility method - not the place to interpret / validate the config. Please move this logic to Server.initHttp() and Server.initHttps(). You will need to add 'error' to the return param list of initHttp().